src/System.Threading.Tasks.Channels/src/System/Threading/Tasks/Channels/Channel.EnumerableChannel.cs

```diff
@@ -0,0 +1,108 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+
+namespace System.Threading.Tasks.Channels
+{
+    public static partial class Channel
+    {
+        /// <summary>Provides a channel around an enumerable.</summary>
+        private sealed class EnumerableChannel<T> : IReadableChannel<T>
+        {
+            /// <summary>A task completed when the channel is done.</summary>
+            private readonly TaskCompletionSource<VoidResult> _completion = new TaskCompletionSource<VoidResult>(TaskCreationOptions.RunContinuationsAsynchronously);
+            /// <summary>The wrapped enumerator.</summary>
+            private readonly IEnumerator<T> _source;
```

Comment:
This enumerator is never disposed. Shouldn't it be disposed when the channel is complete?